test/benchmarks/netperf/README.md

```diff
@@ -0,0 +1,121 @@
+# Benchmarking Kubernetes Networking Performance
+
+## Objectives:
+A standardized benchmark to measure Kubernetes networking performance on multiple host platforms and network stacks.
+
+## Implementation and Test Methodology
+
+The benchmark can be executed via a single Python script invocation that triggers all the automated testing located in the orchestrator and worker pods as seen below. The test uses a custom docker container that has the go binary and iperf3 and other tools built into it. 
+The orchestrator pod coordinates the worker pods to run tests in serial order for the 4 scenarios described below, at MTUs (MSS tuning for TCP and direct packet size tuning for UDP).
+Using node labels, the Worker Pods 1 and 2 are placed on the same Kubernetes node, and Worker Pod 3 is placed on a different node. The nodes all communicate with the orchestrator pod service using simple golang rpcs and request work items.
+
+![](images/BenchmarkingKubernetesNetworkingPerformance.svg)
+
+The 5 major network traffic paths are combination of Pod IP vs Virtual IP and whether the pods are co-located on the same node/VM versus a remotely located pod.
+
+* Same VM using Pod IP
+
+Same VM Pod to Pod traffic tests from Worker 1 to Worker 2 using its Pod IP.
+
+* Same VM using Cluster/Virtual IP
+
+Same VM Pod to Pod traffic tests from Worker 1 to Worker 2 using its Service IP (also known as its Cluster IP or Virtual IP).
+
+* Remote VM using Pod IP
+
+Worker 3 to Worker 2 traffic tests using Worker 2 Pod IP.
+
+* Remote VM using Cluster/Virtual IP
+
+Worker 3 to Worker 2 traffic tests using Worker 2 Cluster/Virtual IP.
+
+* Same VM Pod Hairpin to itself using Cluster IP
+
+The orchestrator and worker pods run independently of the initiator script, with the orchestrator pod sending work items to workers till the testcase schedule is complete.
```

Comment:
what's the minimum number of nodes needed? I'm guessing you don't want the orchestrator also running on the same vm, or is the load from that negligable?